src/components/components/ebay-tooltip-base/index.marko

```diff
@@ -1,3 +1,14 @@
+static function toJSON() {
+    // Only serialize the following to the browser if we are a top level component.
+    return {
```

Comment:
How does this work?
Are these all the variables needed in client side render? So from now on we would have to specify all client side variables needed from server?

Just wondering so i can understand for future

Reply:
It's only needed for split components. If a split component is loaded (outside of a stateful component) it will not rerender in the browser. However, it still gets all of its `input` serialized so that you can access `this.input` inside of the `component-browser.js` file.

By adding the `toJSON` to the `input` it will make it so that if this component is serialized at the top level then it will only send down the subset of `input` that is actually used in the `component-browser.js`. For some of the components, we have no `input` is used in the `component-browser.js`, that's why many `toJSON`'s are a noop, so nothing gets serialized.